File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/Auto/AutoBase.java
package org.firstinspires.ftc.teamcode.Auto;



import com.qualcomm.robotcore.eventloop.opmode.LinearOpMode;
import org.firstinspires.ftc.teamcode.drive.SampleMecanumDrive;
import com.acmerobotics.roadrunner.geometry.Pose2d;
import org.firstinspires.ftc.teamcode.trajectorysequence.TrajectorySequence;
import org.firstinspires.ftc.teamcode.trajectorysequence.TrajectorySequenceBuilder;
import com.qualcomm.robotcore.hardware.HardwareMap;

public abstract class AutoBase extends LinearOpMode {
  
  
  SampleMecanumDrive drive;
  
  Pose2d startPose;
  
  AutoBase(){
      this(new Pose2d(0, 0, Math.toRadians(0)));
  }
  
  AutoBase(Pose2d startPose){
    drive = new SampleMecanumDrive(hardwareMap);
    this.startPose = startPose;
  }
  
  
  
}

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/auto/AutoBase.kt
package org.firstinspires.ftc.teamcode.auto

import com.arcrobotics.ftclib.command.CommandOpMode
import com.arcrobotics.ftclib.command.CommandScheduler
import com.arcrobotics.ftclib.command.RunCommand
import com.arcrobotics.ftclib.command.WaitCommand
import com.qualcomm.hardware.lynx.LynxModule
import org.firstinspires.ftc.teamcode.commands.BulkCacheCommand
import org.firstinspires.ftc.teamcode.drive.SampleMecanumDrive

abstract class AutoBase : CommandOpMode() {

    //convenience command for a pause
    fun waitFor(millis: Long): WaitCommand = WaitCommand(millis);

    //start bulk caching
    override fun initialize() {

        //Clear the bulk read cache every iteration
        schedule(BulkCacheCommand(hardwareMap))
    }

    //update the telemetry in the background as well
    override fun run() {
        super.run()

        telemetry.update()
    }

}
